internal/store/pod.go

```diff
@@ -87,6 +87,34 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_pod_ready_time",
+			Type: metric.Gauge,
+			Help: "Unix timestamp for the moment a pod becomes ready.",
+			GenerateFunc: wrapPodFunc(func(p *v1.Pod) *metric.Family {
+				ms := []*metric.Metric{}
+
+				var readyTime *metav1.Time
+
+				for _, condition := range p.Status.Conditions {
+					if condition.Type == "Ready" {
+						readyTime = &condition.LastTransitionTime
```

Comment:
Hmm this is very approximate value to me, as we don't directly map like we do in start time metric where we can directly look at the value of StartTime. I would if we want to introduce this mark this metric as experimental. 

cc @brancz @tariq1890

Reply:
@lilic should I just add the EXPERIMENTAL flag to the docs? I also wouldn't trust this metric so much right now.

Reply:
Yes go ahead, add EXPERIMENTAL, still waiting for one of the other two maintainers if they agree on adding this.

Reply:
So the issue here is that we are adding custom logic to create a metric. We are trying to the stick to the pattern where kube-state-metrics acts as a mirror of the Kubernetes API objects. Any special logic that needs to be executed should be done via promQL. 

In light of this, I am leaning towards creating a metric like `kube_pod_status_condition_last_transition_time`. This could include the pod status condition type as a label.

Reply:
That could be even better, making the kube-state-metrics side simpler. I'll rewrite it later today.

Reply:
@tariq1890 I removed the other metric and added `kube_pod_status_condition` and `kube_pod_status_condition_last_transition_time`.